astropy/timeseries/downsample.py

```diff
@@ -61,45 +81,83 @@ def aggregate_downsample(time_series, *, time_bin_size=None, time_bin_start=None
     if not isinstance(time_series, TimeSeries):
         raise TypeError("time_series should be a TimeSeries")
 
-    if not isinstance(time_bin_size, u.Quantity):
-        raise TypeError("time_bin_size should be a astropy.unit quantity")
+    if time_bin_size is not None and not isinstance(time_bin_size, (u.Quantity, TimeDelta)):
+        raise TypeError("'time_bin_size' should be a Quantity or a TimeDelta")
 
-    bin_size_sec = time_bin_size.to_value(u.s)
+    if time_bin_start is not None and not isinstance(time_bin_start, (Time, TimeDelta)):
+        time_bin_start = Time(time_bin_start)
+
+    if time_bin_end is not None and not isinstance(time_bin_end, (Time, TimeDelta)):
+        time_bin_end = Time(time_bin_end)
 
     # Use the table sorted by time
     sorted = time_series.iloc[:]
 
+    # The following part makes sure that backwards compatability is provided.
     # Determine start time if needed
     if time_bin_start is None:
         time_bin_start = sorted.time[0]
 
-    # Find the relative time since the start time, in seconds
-    relative_time_sec = (sorted.time - time_bin_start).sec
+    if time_bin_start.isscalar:
+        time_duration = (sorted.time[-1] - time_bin_start).sec
+
+    if time_bin_size is None and time_bin_end is None:
+        if time_bin_start.isscalar:
+            if n_bins is None:
+                raise TypeError("Insufficient binning arguments are provided")
```

Comment:
```suggestion
                raise TypeError("With single 'time_bin_start' either 'n_bins' or 'time_bin_size|end' must be provided")
```
Already a vast improvement over the previous "time_bin_size should be a astropy.unit quantity", but even better give specific diagnostics on which input to fix.